src/mptcpize.c

```diff
@@ -55,6 +58,20 @@ static char doc[] =
         "\tdisable <unit>            Update the systemd <unit>, removing\n"
         "\t                          the above launcher.\n";
 
+#ifndef HAVE_ERROR_H
+# define ERROR_HELPER(status, errnum, format, ...) do {		\
+	if (errnum) {						\
+		errno = errnum;					\
+		perror(format);					\
+	} else {						\
+		fprintf(stderr, format "%s", __VA_ARGS__);	\
+	}							\
+	if (status)						\
+		exit(status);					\
+   } while(0)
+# define error(...) ERROR_HELPER(__VA_ARGS__, "\n")
+#endif
```

Comment:
musl provides the [`err()`](https://git.musl-libc.org/cgit/musl/tree/src/legacy/err.c) function with the same parameter list and behavior. It is defined in [<err.h>](https://git.musl-libc.org/cgit/musl/tree/include/err.h). I think we should fall back on `err()` for musl instead, but I'm not adamant about that. Thoughts?

Reply:
@ossama-othman I didn't know about `err()`, thank you for the tip. I will look at that, looks better and easier.

Reply:
@ossama-othman mmh, it doesn't help that much: `err()` takes the global `errno`, but `error()` is not always used with `errno`. To keep the same message, it looks like we still need to use `perror()`.

Should we keep that like that? Or calling `errno = errnum; err(status, format, __VA_ARGS__)`?

Reply:
> Or calling `errno = errnum; err(status, format, __VA_ARGS__)`?

Mmh, that's not enough: if we call `err(0, ...)`, it will add `: Success` :)
Or maybe we don't need to care?